### PR TITLE
[mqtt] Add edgehub feature to enable EdgeHub integration

### PR DESCRIPTION
**Pull request by dmolokanov**
* added `edgehub` feature that enables code related to communication with EdgeHub
* make `edgehub` features enabled by default
* changed mqtt-images pipeline to create image **without** edgehub integration

### CODE REVIEW

File: mqtt/mqtt-broker-core/src/auth/authentication.rs [hunk 1]
```diff
@@ -48,6 +48,15 @@ pub trait Authenticator {
     ) -> Result<Option<AuthId>, Self::Error>;
 }
 
+/// Creates a authenticator from a function.
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker-core/src/auth/authentication.rs:51]:
```suggestion
/// Creates an authenticator from a function.
```

File: mqtt/mqtt-broker-core/src/auth/authorization.rs [hunk 2]
```diff
@@ -14,6 +14,15 @@ pub trait Authorizer {
     fn authorize(&self, activity: Activity) -> Result<bool, Self::Error>;
 }
 
+/// Creates a authorizer from a function.
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker-core/src/auth/authorization.rs:17]:
```suggestion
/// Creates an authorizer from a function.
```

File: mqtt/mqtt-broker/src/connection.rs [hunk 3]
```diff
@@ -264,17 +266,20 @@ where
...
                     Packet::PubRec(pubrec) => ClientEvent::PubRec(pubrec),
                     Packet::PubRel(pubrel) => ClientEvent::PubRel(pubrel),
                     Packet::Subscribe(subscribe) => {
+                        #[cfg(feature = "edgehub")]
                         let subscribe = translate_incoming_subscribe(&client_id, subscribe);
                         ClientEvent::Subscribe(subscribe)
                     }
                     Packet::SubAck(suback) => ClientEvent::SubAck(suback),
                     Packet::Unsubscribe(unsubscribe) => {
+                        #[cfg(feature = "edgehub")]
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker/src/connection.rs:282]: q: what is the scope of `cfg`, just a following expression? It's hard to find the answer online :(

Comment (dmolokanov) [reply mqtt/mqtt-broker/src/connection.rs:282]: Correct. To wrap more than one expression one need to wrap it with `{...}`

**Pull request merged**